src/gui/processing/qgsprocessinghistoryprovider.cpp

```diff
@@ -72,8 +81,195 @@ void QgsProcessingHistoryProvider::portOldLog()
+    {
+      if ( !mPythonCommand.isEmpty() )
+      {
+        QAction *pythonAction = new QAction(
+          QObject::tr( "Copy as Python Command" ), menu );
+        pythonAction->setIcon( QgsApplication::getThemeIcon( QStringLiteral( "mIconPythonFile.svg" ) ) );
+        QObject::connect( pythonAction, &QAction::triggered, menu, [ = ]
+        {
+          copyText( mPythonCommand );
+        } );
+        menu->addAction( pythonAction );
+      }
+      if ( !mProcessCommand.isEmpty() )
+      {
+        QAction *processAction = new QAction(
+          QObject::tr( "Copy as qgis_process Command" ), menu );
+        processAction->setIcon( QgsApplication::getThemeIcon( QStringLiteral( "mActionTerminal.svg" ) ) );
+        QObject::connect( processAction, &QAction::triggered, menu, [ = ]
+        {
+          copyText( mProcessCommand );
+        } );
+        menu->addAction( processAction );
+      }
+      if ( !mInputs.isEmpty() )
+      {
+        QAction *inputsAction = new QAction(
+          QObject::tr( "Copy as JSON" ), menu );
+        inputsAction->setIcon( QgsApplication::getThemeIcon( QStringLiteral( "mActionEditCopy.svg" ) ) );
+        QObject::connect( inputsAction, &QAction::triggered, menu, [ = ]
+        {
+          copyText( QString::fromStdString( QgsJsonUtils::jsonFromVariant( mInputs ).dump( 2 ) ) );
+        } );
+        menu->addAction( inputsAction );
+      }
+
+      if ( !mPythonCommand.isEmpty() )
+      {
+        if ( !menu->isEmpty() )
+        {
+          menu->addSeparator();
+        }
+
+        QAction *createTestAction = new QAction(
+          QObject::tr( "Create Test…" ), menu );
+        QObject::connect( createTestAction, &QAction::triggered, menu, [ = ]
+        {
+          mProvider->emitCreateTest( mPythonCommand );
+        } );
+        menu->addAction( createTestAction );
+      }
+    }
+
+    void copyText( const QString &text )
+    {
+      QMimeData *m = new QMimeData();
+      m->setText( text );
+      QApplication::clipboard()->setMimeData( m );
+    }
+
+    QgsHistoryEntry mEntry;
+    QString mAlgorithmId;
+    QString mPythonCommand;
+    QString mProcessCommand;
+    QVariantMap mInputs;
+    QString mDescription;
+
+    mutable QgsProcessingAlgorithmInformation mAlgorithmInformation;
+    QgsProcessingHistoryProvider *mProvider = nullptr;
+
+};
+
+QgsHistoryEntryNode *QgsProcessingHistoryProvider::createNodeForEntry( const QgsHistoryEntry &entry, const QgsHistoryWidgetContext & )
+{
+  return new ProcessingHistoryNode( entry, this );
+}
+
 QString QgsProcessingHistoryProvider::oldLogPath() const
 {
   const QString userDir = QgsApplication::qgisSettingsDirPath() + QStringLiteral( "/processing" );
   return userDir + QStringLiteral( "/processing.log" );
 }
+
+void QgsProcessingHistoryProvider::execute( const QString &commands )
+{
+  emit executePython( commands );
+}
+
+void QgsProcessingHistoryProvider::emitCreateTest( const QString &command )
```

Comment:
I've raised it in the header file already but to be consistent here, we should either call this createTest, or rename execute to emitExecute.

I'd favor renaming this function to createTest.